src/appservice/MjolnirManager.ts

```diff
@@ -222,4 +246,56 @@ export class ManagedMjolnir {
     public get managementRoomId(): string {
         return this.mjolnir.managementRoomId;
     }
+
+    /**
+     * Intended to be called by the MjolnirManager to make sure the mjolnir is ready to listen to events.
+     * It should not be informed of any events via `onEvent` until this is called.
+     */
+    public async start(): Promise<void> {
+        await this.mjolnir.start();
+    }
+}
+
+export class MatrixIntentListener extends EventEmitter implements MatrixEmitter {
```

Comment:
Nit: The term "intent" is a bit too much overloaded for my tastes. Maybe something like `AppserviceTransactionListener`?

Also, some documentation would be useful.